server/main.coffee

```diff
@@ -0,0 +1,129 @@
+SPARQurL = process.env.SPARQURL || 'http://localhost:3030/dataset'
+
+Meteor.methods(
+
+  'SPARQurL': () ->
+    SPARQurL
+
+  'init' : () ->
+    JSON.parse
+
+  'getRecentlyMentionedInfectiousAgents' : () ->
+    query = '''
+      prefix xsd: <http://www.w3.org/2001/XMLSchema#>
+      prefix anno: <http://www.eha.io/types/annotation_prop/>
+      prefix rdf: <http://www.w3.org/2000/01/rdf-schema#>
+      prefix promed: <http://www.eha.io/types/promed/>
+      prefix dc: <http://purl.org/dc/terms/>
+      SELECT
+          # For each of the most recently mentioned terms find the most recent
+          # mention prior to the current mention.
+          ?resolvedTerm ?currentDate ?currentArticle
+          (sample(?termLabel) as ?word)
+          (max(?prevArticle) as ?priorArticle)
+          (max(?prevDate) as ?priorDate)
+      WHERE {
+          # Select the most recently mentioned terms.
+          # Doing this as a subquery speeds up the overall query
+          # by limiting items prior mentions are found for.
+          {
+              SELECT DISTINCT ?resolvedTerm ?termLabel ?currentDate ?currentArticle
+              WHERE {
+                  ?phrase anno:category "diseases"
+                      ; anno:source_doc ?currentArticle
+                      ; anno:start ?start
+                      ; ^dc:relation ?resolvedTerm
+                      .
+                  ?resolvedTerm rdf:label ?termLabel .
+                  ?currentArticle promed:date ?currentDate .
+              }
+              # Sort by date, then document, then offset within the document.
+              ORDER BY DESC(?currentDate) DESC(?currentArticle) ASC(?start)
+              LIMIT 50
+          }
+          # Select the previous usages of the most recently mentioned terms
+          OPTIONAL {
+            ?prev_mention anno:source_doc ?prevArticle
+                ; ^dc:relation ?resolvedTerm
+                .
+            ?prevArticle promed:date ?prevDate .
+            FILTER(?currentDate >= ?prevDate && ?currentArticle != ?prevArticle)
+          }
+      }
+      # Group by the items from the inner query
+      GROUP BY ?resolvedTerm ?currentDate ?currentArticle
```

Comment:
The list isn't currently ordered by currentDate;  Adding:
```ORDER BY DESC(?currentDate)```
fixes that, but oddly, the result looks partially alphabetized after doing so.